### ISSUE TITLE
Export subsribers with data source(s)

### ISSUE THREAD

**Issue by iulistadmin** [sympa-community/sympa#1157]
Make it possible to export (i.e., dump) subscribers/members and their data source(s).

Expected Behavior
-----------------
An option to dump members with data sources could be provided in the web interface.

Current Behavior
----------------
This option does not exist. Members may be dumped as email addresses only or with names, if present.

Possible Solution
-----------------
Provide an option to dump members with their data source(s).

Context
-------
Owners would like to have access to this data to assist them with list member management.

**Comment by ikedas** [comment]
`include_sympa_list` or `include_remote_sympa_list` may treat the lists as data sources.  See also [documentation](https://sympa-community.github.io/manual/customize/data-sources.html).

**Comment by iulistadmin** [comment]
Let me clarify. We hide the inclusion options from owners in the web interface. So, only the list administrator may see and edit these options. The --dump_users will export the subscriber information with the inclusion_label data for each member, but that again is only available to the list administrator via command line. Owners have no ability to acquire the data via the web interface, or am I missing something? Thanks.

**Comment by ikedas** [comment]
"Subscriber Table" in `review` (Subscribers) page includes "Sources" column that contains the name of data source if each subscriber is included.

**Comment by iulistadmin** [comment]
Yes. I know. That's not the request. The request is to have a means for the owner to export the subscribers with the source data. Also, the source field only shows one source. So, if a list has multiple sources, for example, two sublists, and the member is on both sublists, there is only one source shown, probably the last one Sympa read from which included that member. That's a separate but related issue. The point of this issue is for the owner(s) to have the ability to include the source(s) data in an export of the list's members.

**Comment by ikedas** [comment]
Please see #1156 about “multiple sources” (I think such thing is impossible).

BTW a subscriber can be both “subscribing” and “included”, or either of them.  What is your idea about how they will be shown in the space-delimited record?

**Comment by iulistadmin** [comment]
Yes. If the member is subscribed directly to the list and the member is also included from a source, both "subscribed" and one data source appear in the Source column. Emphasis on one. If the member is subscribed to more than one source, those are not included in the Subscribers table displayed in the web UI. And there is no export option available to list owners to export the subscribers with their source(s) data therein. As an administrator, the --dump_users option will do this, but owners have no such option, as far as I know.

**Comment by ikedas** [comment]
Then, how do you think of the format of exported data for the owners?

Currently it is simple plain text containing with one person per line:

``` code
user1@example.com Joan Doe
user2@example.org Erika Mustermann
user3@example.net Pierre Paul Jacques
```

If possible, we’d be better to expand this format to include other information such as "subscribing"/"included" labels.

**Comment by racke** [comment]
I would love to see an option for an expanded format using YAML syntax.

**Comment by iulistadmin** [comment]
Sure. That would be fine, or even a simple comma delimited source list where if the subscriber is imported from more than one source, they are shown:

```
user1@example.com Joan Doe subscribed
user2@example.org Erika Mustermann sub-list1,sub-list3
user3@example.net Pierre Paul Jacques MyOrg-LDAP-Group,/path/to/local/file/source/members.txt
```

**Comment by ikedas** [comment]
However, display names (a.k.a. "gecos") of subscribers may contain any characters, such as comma "`,`".  I think that's why @racke suggested the new format such as YAML.  Do you have any idea?

---
NB: If you need the feature to provide information of subscribers to owners right now, you may write a script to run `sympa.pl --dump_users`, modify the dump files reducing unneccesary information and to send them to owners.

**Comment by iulistadmin** [comment]
Oh, good point. Sure, YAML format is fine for the export.

**Comment by racke** [comment]
Yes and we can add other interesting information about the users, e.g.their preferences.
